File: PHP/Rebill/Create_Recurring_Payment_CC.php
<?php
/**
* BluePay PHP Sample Code
*
* This code sample runs a $0.00 Credit Card Auth transaction
* against a customer using test payment information.
* See comments below on the details of the initial setup of the
* rebilling cycle.
*/

include('../BluePay.php');

$accountID = "Merchant's Account ID Here";
$secretKey = "Merchant's Secret Key Here";
$mode = "TEST";

$rebill = new BluePay(
    $accountID,
    $secretKey,
    $mode
);

$rebill->setCustomerInformation(array(
    'firstName' => 'Bob', 
    'lastName' => 'Tester', 
    'addr1' => '1234 Test St.', 
    'addr2' => 'Apt #500', 
    'city' => 'Testville', 
    'state' => 'IL', 
    'zip' =>'54321', 
    'country' => 'USA', 
    'phone' => '1231231234', 
    'email' => 'test@bluepay.com' 
));
 
$rebill->setCCInformation(array(
    'cardNumber' => '4111111111111111', // Card Number: 4111111111111111
    'cardExpire' => '1215', // Card Expire: 12/15
    'cvv2' => '123' // Card CVV2: 123
));

$rebill->setRebillingInformation(array(
   'rebillFirstDate' => '2015-01-05', // Rebill Start Date: Jan. 5, 2015
   'rebillExpression' => '1 MONTH', // Rebill Frequency: 1 MONTH
   'rebillCycles' => '5', // Rebill # of Cycles: 5
   'rebillAmount' => '3.50' // Rebill Amount: $3.50
));

$rebill->auth('0.00');

// Makes the API Request with BluePay
$rebill->process();

if ($rebill->isSuccessfulResponse()) {
    // Read response from BluePay
    echo 
    'Transaction ID: '. $rebill->getTransID() . "\n" .
    'Rebill ID: ' . $rebill->getRebillID() . "\n" .
    'Status: '. $rebill->getStatus() . "\n" .
    'Message: '. $rebill->getMessage() . "\n" .
    'AVS Response: ' . $rebill->getAVSResponse() . "\n" .
    'CVS Response: ' . $rebill->getCVV2Response() . "\n" .
    'Masked Account: ' . $rebill->getMaskedAccount() . "\n" .
    'Card Type: ' . $rebill->getCardType() . "\n" .
    'Authorization Code: ' . $rebill->getAuthCode() . "\n";
} else {
    echo $rebill->getMessage();
}

?>

File: C++/Rebill/Create_Recurring_Payment_CC.cpp
//
// BluePay C++ Sample code.
//
// This code sample creates a recurring payment charging $15.00 per month for one year.

#include "Create_Recurring_Payment_CC.h"
#include "../bluepay-cpp/BluePay.h"
#include <iostream>
using namespace std;

void createRecurringPaymentCC(){

    string accountId = "Merchant's Account ID Here";
    string secretKey = "Merchant's Secret Key Here";
    string mode = "TEST";

    BluePay rebill(
    	accountId, 
	    secretKey, 
	    mode 
    );

    rebill.setCustomerInformation(
	    "Bob", // First Name
	    "Tester", // Last Name
	    "1234 Test St.", // Address1
	    "Apt #500", // Address2
	    "Testville", // City
	    "IL", // State
	    "54321", // Zip
	    "USA", // Country
	    "1231231234", // Phone Number
	    "test@bluepay.com" // Email Address
	  );

    rebill.setCCInformation(
	    "4111111111111111", // Card Number
	    "1225", // Card Expire
	    "123" // Card CVV2
    );

	rebill.setRebillingInformation(
		"15.00", // Rebill Amount
		"2015-01-01", // Rebill Start Date
		"1 MONTH", // Rebill Frequency
		"12" // Rebill number of Cycles
	);

	// Auth Amount: $0.00
	rebill.auth("0.00");
    
    // Makes the API Request with Blue
	rebill.process();

	// Reads the responses from BluePay if transaction was approved
	if (rebill.isSuccessfulTransaction()){
	    cout << string("Transaction ID: ") + rebill.getTransId() + "\n";
		cout << string("Rebill ID: ") + rebill.getRebillId() + "\n";
	    cout << string("Transaction Status: ") + rebill.getResult() + "\n";
	    cout << string("Transaction Message: ") + rebill.getMessage() + "\n";
	    cout << string("AVS Result: ") + rebill.getAvs() + "\n";
	    cout << string("CVV2 Result: ") + rebill.getCvv2() + "\n";
	    cout << string("Masked Payment Account: ") + rebill.getMaskedPaymentAccount() + "\n";
	    cout << string("Card Type: ") + rebill.getCardType() + "\n";
	    cout << string("Authorization Code: ") + rebill.getAuthCode() + "\n";
	}
	else {
	    cout << string("Error: ") + rebill.getMessage();
	}

}